crates/control/src/role_assignment.rs

```diff
@@ -211,7 +211,14 @@ impl RoleAssignment {
 
         let mut team_ball = self.team_ball;
 
-        if spl_striker_message_timeout {
+        let is_in_penalty_kick = matches!(
+            context.filtered_game_controller_state,
+            Some(FilteredGameControllerState {
+                sub_state: Some(SubState::PenaltyKick),
+                ..
+            })
+        );
+        if spl_striker_message_timeout && !is_in_penalty_kick {
```

Comment:
isnt this a band-aid fix and we want proper path-planning instead?

Reply:
A little bit? As far as I can tell the problem is that searchers ignore obstacles in path planning.
On the other hand I have never seen a robot become searcher in a penalty kick, which means that striker message timeout could be a behavior simulator specific bug. See #1205